content/en/docs/reference/command-line-tools-reference/kubelet-tls-bootstrapping.md

```diff
@@ -197,19 +370,74 @@ roleRef:
   apiGroup: rbac.authorization.k8s.io
 ```
 
+
+The `csrapproving` controller that ships as part of
+[kube-controller-manager](/docs/admin/kube-controller-manager/) and is enabled
+by default. The controller uses the [`SubjectAccessReview`
+API](/docs/reference/access-authn-authz/authorization/#checking-api-access) to
+determine if a given user is authorized to request a CSR, then approves based on
+the authorization outcome. To prevent conflicts with other approvers, the
+builtin approver doesn't explicitly deny CSRs. It only ignores unauthorized
+requests. The controller also prunes expired certificates as part of garbage
+collection.
+
+
 ## kubelet configuration
+Finally, with the master nodes properly set up and all of the necessary authentication and authorization in place, we can configure the kubelet.
+
+The kubelet requires the following configuration to bootstrap:
 
-To request a client certificate from kube-apiserver, the kubelet first needs a
-path to a kubeconfig file that contains the credentials for the identity that it
-will use to bootstrap its individual node identity.
+* A path to store the key and certificate it generates (optional, can use default)
+* A path to a `kubeconfig` file that does not yet exist; it will place the bootstrapped config file here
+* A path to a bootstrap `kubeconfig` file to provide the bootstrap token and URL for the server
+* Optional: instructions to rotate certificates
 
-If you are using a bootstrap token, you can use `kubectl config set-cluster`,
-`set-credentials`, and `set-context` to build this kubeconfig.  Provide the name
-`kubelet-bootstrap` to `kubectl config set-credentials` and include
-`--token=<token-value>` as follows:
+The bootstrap `kubeconfig` should be in a path available to the kubelet, for example `/var/lib/kubelet/bootstrap-kubeconfig`.
 
+Its format is identical to a normal `kubeconfig` file. A sample file might look as follows:
+
+```yml
+apiVersion: v1
+clusters:
+- cluster:
+    certificate-authority: /var/lib/kubernetes/ca.pem
+    server: https://my.server.example.com:6443
+  name: bootstrap
+contexts:
+- context:
+    cluster: bootstrap
+    user: kubelet-bootstrap
+  name: bootstrap
+current-context: bootstrap
+kind: Config
+preferences: {}
+users:
+- name: kubelet-bootstrap
+  user:
+    token: 07401b.f395accd246ae52d
 ```
-kubectl config set-credentials kubelet-bootstrap --token=${BOOTSTRAP_TOKEN} --kubeconfig=bootstrap.kubeconfig
+
+The important elements to note are:
+
+* `certificate-authority`: path to a CA file, used to validate the server certificate presented by kube-apiserver
+* `server`: URL to kube-apiserver
+* `token`: the token to use
+
+The format of the token does not matter, as long as it matches what kube-apiserver expects. In the above example, we used a bootstrap token.
```

Comment:
note here that any valid authentication method can be used, not just tokens

Reply:
👍